File: easy/validate_subsequence.rb
def isValidSubsequence(array, sequence)
  arrIdx = 0
  seqIdx = 0
  while arrIdx < array.length and seqIdx < sequence.length
    if sequence[seqIdx] == array[arrIdx]
      seqIdx += 1
    end
    arrIdx += 1
  end
  if seqIdx == sequence.length
    return true
  end
  false
end

array = [5, 1, 22, 25, 6, -1, 8, 10]
sequence = [1, 6, -1, 10]

print isValidSubsequence(array, sequence)

File: easy/validate-subsequence.py
def isValidSubsequence(array, sequence):
    arr_idx = 0
    seq_idx = 0
# iterate through the array looking for the first number in our sequence
# when we find it, we want to increment the sequence index
# if we find all numbers in the sequence, return true
# if seqIdx = len(sequence)
# if we make it through the array without finding all of the numbers
#   return false
# if arrIdx = len(array) and seqIdx < len(sequence)
    while arr_idx < len(array) and seq_idx < len(sequence):
        if sequence[seq_idx] == array[arr_idx]:
            seq_idx += 1
        arr_idx += 1
    if seq_idx == len(sequence):
        return True
    return False
